src/ui/components/ArchivedCredentials/ArchivedCredentials.test.tsx

```diff
@@ -0,0 +1,220 @@
+      "Select"
+    );
+  });
+
+  test("Restore multiple archived credentials", async () => {
+    const { getByText, getByTestId } = render(
+      <Provider store={mockedStore}>
+        <ArchivedCredentialsContainer
+          archivedCredentialsIsOpen={true}
+          archivedCreds={credsFixW3c}
+          setArchivedCredentialsIsOpen={jest.fn()}
+        />
+      </Provider>
+    );
+
+    await waitFor(() => {
+      expect(getByTestId("action-button")).toBeVisible();
+    });
+
+    act(() => {
+      fireEvent.click(getByText("Select"));
+    });
+
+    await waitFor(() => {
+      expect(getByTestId("action-button").children.item(0)?.innerHTML).toBe(
+        "Cancel"
+      );
+    });
+
+    await waitFor(() => {
+      expect(getByText("0 Credentials Selected")).toBeVisible();
+    });
+
+    const cardItem = getByTestId(`crendential-card-item-${credsFixW3c[0].id}`);
+    fireEvent.click(cardItem);
+
+    const cardItem1 = getByTestId(`crendential-card-item-${credsFixW3c[1].id}`);
+    fireEvent.click(cardItem1);
+
+    const cardItem2 = getByTestId(`crendential-card-item-${credsFixW3c[2].id}`);
+    fireEvent.click(cardItem2);
+
+    await waitFor(() => {
+      expect(
+        getByText(`${credsFixW3c.length} Credentials Selected`)
+      ).toBeVisible();
+    });
+
+    fireEvent.click(getByTestId("restore-credentials"));
+
+    await waitFor(() => {
+      expect(
+        getByText(EN_TRANSLATIONS.creds.card.details.alert.restore.confirm)
+      ).toBeVisible();
+    });
+
+    fireEvent.click(
+      getByText(EN_TRANSLATIONS.creds.card.details.alert.restore.confirm)
+    );
+
+    await waitFor(() => {
+      expect(dispatchMock).toBeCalledWith(setCredsCache([...credsFixW3c]));
+    });
+  });
+
+  test.skip("Delete multiple archived credential", async () => {
+    const { getByText, getByTestId } = render(
+      <Provider store={mockedStore}>
+        <ArchivedCredentialsContainer
+          archivedCredentialsIsOpen={true}
+          archivedCreds={credsFixW3c}
+          setArchivedCredentialsIsOpen={jest.fn()}
+        />
+      </Provider>
+    );
+
+    await waitFor(() => {
+      expect(getByTestId("action-button")).toBeVisible();
+    });
+
+    act(() => {
+      fireEvent.click(getByText("Select"));
+    });
+
+    await waitFor(() => {
+      expect(getByTestId("action-button").children.item(0)?.innerHTML).toBe(
+        "Cancel"
+      );
+
+      expect(getByTestId("delete-credentials")).toBeVisible();
+    });
+
+    await waitFor(() => {
+      expect(getByText("0 Credentials Selected")).toBeVisible();
+    });
+
+    const cardItem = getByTestId(`crendential-card-item-${credsFixW3c[0].id}`);
+
+    fireEvent.click(cardItem);
+
+    fireEvent.click(getByTestId("delete-credentials"));
+
+    await waitFor(() => {
+      expect(
+        getByText(EN_TRANSLATIONS.creds.card.details.alert.delete.confirm)
+      ).toBeVisible();
+    });
```

Comment:
As discussed in DM, please cut it short and just test that when you select multiple credentials and click delete you are asked to confirm the password.

The modal will need to be tested separately as well as the function for deleting the credential (was done in core).